pkg/kratos/service.go

```diff
@@ -355,12 +354,24 @@ func (s *Service) UpdateLoginFlow(
 		return nil, nil, err
 	}
 
+	c := resp.Cookies()
+	if body.UpdateLoginFlowWithOidcMethod != nil {
+		// If this is an oidc flow, we need to delete the session cookie
+		// A session cookie (probably) means that the user used 1fa, but went back from the 2nd factor screen
+		// If the session cookie is set, then Kratos will redirect the user to the default return to URL
+		// The only way to avoid this is by setting refresh=true, but the user has no kratos session.
+		// This is probably a bug on kratos side, as they check if a session exists to set refresh=false.
+		// But in oidc they only check if the session cookie exists, which is not sufficient as the user may not have
+		// enough aal
+		c = append(c, deleteKratosCookie())
```

Comment:
question: so what you're saying is that we need to do this here because kratos doesn't handle the refresh=true situation correctly, right?

The surprising thing for me is the check on the `aal`, I was convinced they did check for it when working on mfa enforcement

Reply:
Let's talk about this, my "feeling" is that it should not be necessary, but I may be day-dreaming

Reply:
see https://github.com/ory/kratos/blob/master/selfservice/strategy/oidc/strategy.go#L344-L356 vs https://github.com/ory/kratos/blob/master/selfservice/flow/login/handler.go#L165

let's discuss after the daily